src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,162 @@
+    }
+
+    public int executeCommand(String input) throws DukeException{
+        String command;
+        String description = null;
+        String time = null;
+        String temp;
+        if (input.contains(" ")) {
+            command = input.substring(0, input.indexOf(" "));
+            temp = input.substring(input.indexOf(" ") + 1);
+            if (temp.contains("/")) {
+                description = input.substring(input.indexOf(" ") + 1, input.indexOf("/") - 1);
+                time = input.substring(input.indexOf("/") + 4);
+            } else {
+                description = temp;
+            }
+        } else {
+            command = input;
+        }
+        switch(command){
+        case "list":
+            if (list.isEmpty()) {
+                throw new NullPointerException();
+            }
+            for(Task task: list){
+                task.printTask();
+            }
+            break;
+        case "done":
+            if (description==null) {
+                throw new NumberFormatException();
+            }
+            int index = Integer.parseInt(description)-1;
+            list.get(index).markAsDone();
+            System.out.println("Nice! I've marked this task as done:");
+            System.out.println("[" + list.get(index).getStatusIcon() + "] " + list.get(index).description);
+            break;
+        case "todo":
+            if (description==null) {
+                throw new DukeException("nullTodo");
+            }
+            count++;
+            Todo todo = new Todo(description, count);
+            printAdded(todo,count);
+            list.add(todo);
+            break;
+        case "deadline":
+            if (description==null) {
+                throw new DukeException("nullTodo");
+            } else if (time==null) {
+                throw new DukeException("nullTime");
+            }
+            count++;
+            Deadline deadline = new Deadline(description, count, time);
+            printAdded(deadline,count);
+            list.add(deadline);
+            break;
+        case "event":
+            if (description==null) {
+                throw new DukeException("nullTodo");
+            } else if (time==null) {
+                throw new DukeException("nullTime");
+            }
+            count++;
+            Event event = new Event(description, count, time);
+            printAdded(event,count);
+            list.add(event);
+            break;
+        case "bye":
+            System.out.println("Bye. Hope to see you again soon!");
+            return -1;
+        case "delete":
+            if (description==null) {
+                throw new NumberFormatException();
+            }
+            int index2 = Integer.parseInt(description)-1;
+            list.get(index2).printTaskDelete(count-1);
+            list.remove(index2);
+            count--;
+            for (int i=index2; i<count; i++) {
+                list.get(i).index--;
+            }
```

Comment:
For each command, consider processing the command in a separate method.

Reply:
Noted